packages/http/src/client.ts

```diff
@@ -0,0 +1,141 @@
+
+    const httpUrl: IHttpUrl = {
+      baseUrl: parsedUrl.host ? `${parsedUrl.protocol}//${parsedUrl.host}` : undefined,
+      path: parsedUrl.pathname,
+      query: parseQueryString(parsedUrl.query || ''),
+    };
+
+    const data = await obj.request(
+      {
+        ...input,
+        url: httpUrl,
+      },
+      resources,
+      config,
+    );
+
+    const output: PrismOutput = {
+      status: data.output.statusCode,
+      statusText: pipe(
+        data.output.statusText,
+        getOrElse(() => getStatusText(data.output.statusCode)),
+      ),
+      headers: data.output.headers || {},
+      data: data.output.body || {},
+      config: defaults(config, defaultConfig),
+      request: { ...input, url: httpUrl },
+      validations: data.validations,
+    };
+
+    return output;
+  };
+
+  function isInput(input?: Pick<IHttpRequest, 'headers'> | IHttpConfig): input is Pick<IHttpRequest, 'headers'> {
+    return !!input && 'headers' in input;
+  }
+
+  return {
+    request,
+    get: (url: string, input?: Pick<IHttpRequest, 'headers'> | IHttpConfig, config?: IHttpConfig) =>
+      isInput(input) ? request(url, { method: 'get', ...input }, config) : request(url, { method: 'get' }, input),
+    put: (url: string, body: unknown, input?: Pick<IHttpRequest, 'headers'> | IHttpConfig, config?: IHttpConfig) =>
+      isInput(input)
+        ? request(url, { method: 'put', ...input, body }, config)
+        : request(url, { method: 'put', body }, input),
+    post: (url: string, body: unknown, input?: Pick<IHttpRequest, 'headers'> | IHttpConfig, config?: IHttpConfig) =>
+      isInput(input)
+        ? request(url, { method: 'post', ...input, body }, config)
+        : request(url, { method: 'post', body }, input),
+    delete: (url: string, input?: Pick<IHttpRequest, 'headers'> | IHttpConfig, config?: IHttpConfig) =>
+      isInput(input) ? request(url, { method: 'delete', ...input }, config) : request(url, { method: 'delete' }, input),
+    options: (url: string, input?: Pick<IHttpRequest, 'headers'> | IHttpConfig, config?: IHttpConfig) =>
+      isInput(input)
+        ? request(url, { method: 'options', ...input }, config)
+        : request(url, { method: 'options' }, input),
+    head: (url: string, input?: Pick<IHttpRequest, 'headers'> | IHttpConfig, config?: IHttpConfig) =>
+      isInput(input) ? request(url, { method: 'head', ...input }, config) : request(url, { method: 'head' }, input),
+    patch: (url: string, body: unknown, input?: Pick<IHttpRequest, 'headers'> | IHttpConfig, config?: IHttpConfig) =>
+      isInput(input)
+        ? request(url, { method: 'patch', ...input, body }, config)
+        : request(url, { method: 'patch', body }, input),
+    trace: (url: string, input?: Pick<IHttpRequest, 'headers'> | IHttpConfig, config?: IHttpConfig) =>
+      isInput(input) ? request(url, { method: 'trace', ...input }, config) : request(url, { method: 'trace' }, input),
```

Comment:
If you're wondering why this — this is to offer the overloads in TypeScript. It's a little bit of code but it's worth for the developer experience, I'd say.